matplotlib-backend-kitty/__init__.py

```diff
@@ -31,29 +85,23 @@ def f(*args, output=True, **kwargs):
         return f
 
     def show(self):
-
-        icat = __class__._run('kitty', '+kitten', 'icat')
-
         if os.environ.get('MPLBACKEND_KITTY_SIZING', 'automatic') != 'manual':
 
-            tput = __class__._run('tput')
-
             # gather terminal dimensions
-            rows = int(tput('lines'))
-            px = icat('--print-window-size')
-            px = list(map(int, px.split('x')))
+            term_height_px, term_width_px = term_size_px()
 
-            # account for post-display prompt scrolling
-            # 3 line shift for [\n, <matplotlib.axes…, >>>] after the figure
-            px[1] -= int(3*(px[1]/rows))
+            # gather figure dimensions
```

Comment:
Didn't really see the point of the post-display prompt scrolling. Can change it back if the maintainers insist.